var/spack/repos/builtin/packages/msvc/package.py

```diff
@@ -44,23 +62,228 @@ def determine_variants(cls, exes, version_str):
         extras["compilers"]["c"] = extras["compilers"]["cxx"]
         return spec, extras
 
+    def setup_dependent_build_environment(self, env, dependent_spec):
```

Comment:
I think @johnwparent should look at the msvc logic here, and it would also be good if he could test out the branch.